### PR TITLE
Remove dead code related to auditing

### PR DESCRIPTION
**Pull request by albertzaharovits**
This PR removes:
 * the `CompositeAuditTrail` which is designed to fan-out auditing events to multiple logger implementation types. This is not needed because since v7.0 there's only one audit logger implementation, the logfile.
 * any traces of the index-based logger implementation, namely the permission of the internal `_xpack` user to read the audit log index.

Related: https://github.com/elastic/elasticsearch/pull/37707

### CODE REVIEW

**Comment by albertzaharovits** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows-sample

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java [hunk 1]
```diff
@@ -21,29 +22,27 @@
...
-    private final CompositeAuditTrail compositeAuditTrail;
+    private final @Nullable AuditTrail auditTrail;
     private final XPackLicenseState licenseState;
     private final Duration minLogPeriod = Duration.ofMinutes(30);
     protected AtomicReference<Instant> nextLogInstantAtomic = new AtomicReference<>(Instant.EPOCH);
 
-    public AuditTrailService(List<AuditTrail> auditTrails, XPackLicenseState licenseState) {
-        this.compositeAuditTrail = new CompositeAuditTrail(Collections.unmodifiableList(auditTrails));
+    public AuditTrailService(@Nullable AuditTrail auditTrail, XPackLicenseState licenseState) {
+        this.auditTrail = auditTrail;
```
Comment (ywangd) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java:38]: Should we use `NOOP_AUDIT_TRAIL` when the pass-in auditTrail is `null`? I think it's better if we don't have to deal with null within the class.

**Review by ywangd — APPROVED** [review]
LGTM

Linking the PR (#37707) that removed the index audit here for reference.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java (hunk 1 above)
Comment (albertzaharovits) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java:38]: It's definitely better to not have to deal with `null`s. Problem is that `null` here means that auditing is not enabled, while `NOOP` is intended to be used when license level does not allow auditing.
I prefer to not have to deal with the two meanings using the same logger just now...
But thanks for the good point!

**Comment by albertzaharovits** [comment]
Thank you for the review Yang!

**Pull request merged**